cardinal/ecs/engine.go

```diff
@@ -92,6 +93,78 @@ type Engine struct {
 	shutdownMutex sync.Mutex
 }
 
+func (e *Engine) ConsumeEVMMsgResult(evmTxHash string) ([]byte, []error, string, bool) {
+	rcpt, exists := e.consumeEVMMsgResult(evmTxHash)
+	return rcpt.ABIResult, rcpt.Errs, rcpt.EVMTxHash, exists
+}
+
+func (e *Engine) HandleEVMQuery(name string, abiRequest []byte) ([]byte, error) {
+	qry, err := e.GetQueryByName(name)
+	if err != nil {
+		return nil, err
+	}
+	req, err := qry.DecodeEVMRequest(abiRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	reply, err := qry.HandleQuery(NewReadOnlyEngineContext(e), req)
+	if err != nil {
+		return nil, err
+	}
+
+	return qry.EncodeEVMReply(reply)
+}
+
+func (e *Engine) GetEVMMsgResult(evmTxHash string) (EVMTxReceipt, bool) {
+	return e.consumeEVMMsgResult(evmTxHash)
+}
+
+func (e *Engine) GetMessageByName(s string) (message.Message, bool) {
+	for _, msg := range e.registeredMessages {
+		if msg.Name() == s {
+			return msg, true
+		}
+	}
+	return nil, false
+}
+
+func (e *Engine) GetPersonaForEVMAddress(addr string) (string, error) {
```

Comment:
Can we consolidate it with this? https://github.com/Argus-Labs/world-engine/blob/main/cardinal/ecs/persona.go#L255-L283

Both of them is EVM addresses, it's on the same curve and same signing scheme.

Reply:
the function you linked is for matching the `Signer` <> `PersonaTag` pair. that is different than what is going on here, where im looking for the persona that has this EVM address under its authorized address list. so not really anything to do with the signing/curve stuff, moreso with what im looking for in the `SignerComponent`. 

However looking here again, we should probably require `Router` callers from EVM smart contracts to specify the PersonaTag they want to send the message as, and then check if they are actually under that tag's `AuthorizedAddress` list. but that will be a much larger refactor, and shouldn't be done in this PR.